tests/tracking/fluent/test_fluent.py

```diff
@@ -799,6 +799,27 @@ def test_search_runs_no_arguments(search_runs_output_format):
         mlflow.tracking.fluent._get_experiment_id.assert_called_once()
 
 
+def test_search_runs_all_experiments(search_runs_output_format):
+    """
+    When no experiment ID is specified but flag is passed, it should search all experiments.
+    """
+    from mlflow.entities import Experiment
+
+    mock_experiment_id = mock.Mock()
+    mock_experiment = mock.Mock(Experiment)
+    experiment_id_patch = mock.patch(
```

Comment:
Do you need this patch?

Reply:
Yes, to test that _get_experiment_id is not called.